Java build fails with schemaToBind is null issue ( version 6.12.3 onwards)

### Expected Behavior

Java build should be completed successfully

when i use 6.12.2, everything works fine.

### Actual Behaviour

Note: Generating OpenAPI Documentation

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> java.lang.NullPointerException: Cannot invoke "io.swagger.v3.oas.models.media.Schema.getExample()" because "schemaToBind" is null

### Steps To Reproduce

update the  below dependency 
annotationProcessor(group: 'io.micronaut.openapi', name: 'micronaut-openapi', version: '6.12.3')

### Environment Information

JDK 17

### Example Application

_No response_

### Version

4.6.+

--- a/openapi/src/main/java/io/micronaut/openapi/visitor/SchemaDefinitionUtils.java
+++ b/openapi/src/main/java/io/micronaut/openapi/visitor/SchemaDefinitionUtils.java
@@ -1048,9 +1048,14 @@ public final class SchemaDefinitionUtils {
             JsonNode items = objNode.remove(PROP_SCHEMA);
             if (items != null && schemaToBind != null && (schemaToBind.getType() != null && schemaToBind.getType().equals(TYPE_ARRAY))) {
                 try {
-                    schemaToBind.items(Utils.getJsonMapper().readerForUpdating(schemaToBind.getItems()).readValue(items));
+                    Schema<?> updatedSchema = Utils.getJsonMapper().readerForUpdating(schemaToBind.getItems()).readValue(items);
+                    if (updatedSchema != null) {
+                        schemaToBind.items(updatedSchema);
+                    } else {
+                        warn("Error reading Swagger Schema for element [" + element + "] (items schema): " + items, context, element);
+                    }
                 } catch (IOException e) {
-                    warn("Error reading Swagger Schema for element [" + element + "]: " + e.getMessage(), context, element);
+                    warn("Error reading Swagger Schema for element [" + element + "] (items schema): " + e.getMessage(), context, element);
                 }
             }
         }
@@ -2623,7 +2628,12 @@ public final class SchemaDefinitionUtils {
         // need to set placeholders to set correct values and types to example field
         schemaJson = resolvePlaceholders(schemaJson, s -> expandProperties(s, getExpandableProperties(context), context));
         try {
-            schemaToBind = Utils.getJsonMapper().readerForUpdating(schemaToBind).readValue(schemaJson);
+            Schema<?> updatedSchema = Utils.getJsonMapper().readerForUpdating(schemaToBind).readValue(schemaJson);
+            if (updatedSchema != null) {
+                schemaToBind = updatedSchema;
+            } else {
+                warn("Error reading Swagger Schema for element [" + element + "]: " + schemaJson, context, element);
+            }
         } catch (IOException e) {
             warn("Error reading Swagger Schema for element [" + element + "]: " + e.getMessage(), context, element);
         }

--- a/openapi/src/test/groovy/io/micronaut/openapi/visitor/OpenApiControllerVisitorSpec.groovy
+++ b/openapi/src/test/groovy/io/micronaut/openapi/visitor/OpenApiControllerVisitorSpec.groovy
@@ -2497,4 +2497,105 @@ class MyBean {}
         paths."/server-context-path/local-path/test/save/{id}"
         paths."/server-context-path/local-path/test/save/{id}".get
     }
+
+    void "test parameter with incorrect schema type"() {
+
+        buildBeanDefinition('test.MyBean', '''
+package test;
+
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Get;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.inject.Singleton;
+
+@Controller("/path/to/controller/")
+class ControllerExternal {
+
+    @Get(uri = "user/get2", produces = MediaType.APPLICATION_JSON)
+    @Operation(
+          summary = "This operation is used to test OpenAPI definitions",
+          description =
+                  "In this we call we fake a request to get a user")
+    @ApiResponse(
+          responseCode = "200",
+          description = "When the request is successful",
+          content =
+          @Content(
+                  mediaType = "application/json",
+                  schema = @Schema(implementation = User.class)))
+    @ApiResponse(responseCode = "400", description = "Bad params")
+    @ApiResponse(responseCode = "500", description = "Any other exception that cannot be resolved.")
+    @Tag(name = "getUser2")
+    public HttpResponse<String> getUser2(
+          @Parameter(
+                  name = "id",
+                  description = "Unique identifier",
+                  required = true,
+                  schema = @Schema(
+                          implementation = String.class,
+                          type = "query"
+                  )
+          ) String id) {
+        return HttpResponse.ok();
+      }
+
+}
+
+class User {
+    public String id;
+    public String firstName;
+    public String lastName;
+    public String email;
+    @Schema(hidden=true)
+    public String address;
+    @Schema(hidden=true)
+    public String city;
+    @Schema(hidden=true)
+    public String state;
+    @Schema(hidden=true)
+    public String zipcode;
+    @Schema(hidden=true)
+    public String country;
+    @Schema(hidden=true)
+    public String phoneNumber;
+}
+
+class GetUserClientRequest {
+    private String id;
+
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+}
+
+@Singleton
+class MyBean {}
+''')
+        when:
+        OpenAPI openAPI = Utils.testReference
+        def paths = openAPI.paths
+
+        then:
+        paths
+        paths."/path/to/controller/user/get2"
+        paths."/path/to/controller/user/get2".get
+        paths."/path/to/controller/user/get2".get.parameters
+        paths."/path/to/controller/user/get2".get.parameters[0]
+        paths."/path/to/controller/user/get2".get.parameters[0].name == "id"
+        paths."/path/to/controller/user/get2".get.parameters[0].required
+        paths."/path/to/controller/user/get2".get.parameters[0].description == "Unique identifier"
+        paths."/path/to/controller/user/get2".get.parameters[0].schema
+        paths."/path/to/controller/user/get2".get.parameters[0].schema.type == "string"
+    }
 }

